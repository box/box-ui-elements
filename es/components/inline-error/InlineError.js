function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import * as React from 'react';
import InlineNotice from '../inline-notice';
const InlineError = props => /*#__PURE__*/React.createElement(InlineNotice, _extends({}, props, {
  type: "error"
}));
export default InlineError;
//# sourceMappingURL=InlineError.js.map