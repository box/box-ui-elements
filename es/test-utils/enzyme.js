function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({
  children
}) => {
  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: defaultTheme
  }, children);
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
//# sourceMappingURL=enzyme.js.map