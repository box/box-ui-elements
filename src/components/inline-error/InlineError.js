// @flow
import * as React from 'react';
import InlineNotice from '../inline-notice';

type Props = {
    children: React.Node,
    className?: string,
    /** Title of the inline error. */
    title: React.Node,
};

const InlineError = (props: Props) => <InlineNotice {...props} type="error" />;

export default InlineError;
