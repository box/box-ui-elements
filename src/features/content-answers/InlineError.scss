@import '../../styles/variables';

.bdl-InlineError {
    margin-top: $bdl-grid-unit * 4;

    .bdl-InlineError-icon {
        display: flex;
        align-items: center;
        justify-content: center;
        width: $bdl-grid-unit * 8;
        height: $bdl-grid-unit * 8;
        background: $bdl-light-blue-20;
        border-radius: 50%;
    }

    .bdl-InlineError-text {
        flex-grow: 1;
        margin: 0 $bdl-grid-unit * 3;
        font-size: $bdl-fontSize--dejaBlue;
    }

    .bdl-InlineError-alertIcon {
        width: 20px;
        height: 20px;
    }

    .bdl-Media-figure {
        margin: $bdl-grid-unit * 4 $bdl-grid-unit * 3 0 0;
    }

    .bdl-Media-body {
        display: flex;
        align-items: center;
        padding: $bdl-grid-unit * 3 $bdl-grid-unit * 4;
        background-color: $bdl-watermelon-red-10;
        border: 2px solid $bdl-watermelon-red-50;
        border-radius: $bdl-grid-unit * 3;
    }
}
