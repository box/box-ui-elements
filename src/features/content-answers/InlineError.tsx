import * as React from 'react';
import { FormattedMessage } from 'react-intl';

import BoxAi from '../../icon/line/BoxAi';
import Media from '../../components/media';
import TriangleAlert16 from '../../icon/line/TriangleAlert16';

import messages from './messages';

import './InlineError.scss';

const InlineError = () => {
    return (
        <Media className="bdl-InlineError" data-testid="InlineError">
            <Media.Figure>
                <div className="bdl-InlineError-icon">
                    <BoxAi height={16} width={16} />
                </div>
            </Media.Figure>
            <Media.Body>
                <div className="bdl-InlineError-alertIcon">
                    <TriangleAlert16 height={20} width={20} />
                </div>
                <div className="bdl-InlineError-text">
                    <FormattedMessage {...messages.inlineErrorText} />
                </div>
            </Media.Body>
        </Media>
    );
};

export default InlineError;
