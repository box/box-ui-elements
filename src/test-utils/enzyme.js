/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({ children }) => {
    return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>;
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, { wrappingComponent: Wrappers, ...options });

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, { wrappingComponent: Wrappers, ...options });

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
