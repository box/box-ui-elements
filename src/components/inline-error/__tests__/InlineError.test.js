import React from 'react';
import { render } from 'enzyme';

import InlineError from '..';

describe('components/inline-error/InlineError', () => {
    test('should correctly render', () => {
        const children = 'this is a message to put in the notice';
        const title = 'this is the title';
        const className = 'some-class';

        const wrapper = render(
            <InlineError title={title} className={className}>
                {children}
            </InlineError>,
        );

        expect(wrapper).toMatchSnapshot();
    });
});
