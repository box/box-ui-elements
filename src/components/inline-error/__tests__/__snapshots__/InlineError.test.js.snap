// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/inline-error/InlineError should correctly render 1`] = `
<div
  class="some-class inline-alert inline-alert-visible inline-alert-error"
>
  <strong>
    this is the title
  </strong>
  <div>
    this is a message to put in the notice
  </div>
</div>
`;
